src/core/Akka.Persistence.Tests/Fsm/PersistentFSMSpec.cs

```diff
@@ -522,16 +675,63 @@ protected override bool ReceiveCommand(object message)
         {
             return true;
         }
+
+        public static Props Props(string persistenceId, IActorRef client)
+        {
+            return Actor.Props.Create(() => new PersistentEventsStreamer(persistenceId, client));
+        }
     }
 
     #region Custome States
 
-    internal enum UserState
+    public interface IUserState : PersistentFSM.IFsmState { }
+
+    public class Shopping : IUserState
     {
-        Shopping,
-        Inactive,
-        Paid,
-        LookingAround
+        public static Shopping Instance { get; } = new Shopping();
+
+        private Shopping() { }
+        public override bool Equals(object obj) => !ReferenceEquals(obj, null) && obj is Shopping;
+        public override int GetHashCode() => nameof(Shopping).GetHashCode();
+        public override string ToString() => nameof(Shopping);
+
+        public string Identifier => "Shopping";
+    }
+
+    public class Inactive : IUserState
+    {
+        public static Inactive Instance { get; } = new Inactive();
+
+        private Inactive() { }
+        public override bool Equals(object obj) => !ReferenceEquals(obj, null) && obj is Inactive;
+        public override int GetHashCode() => nameof(Inactive).GetHashCode();
+        public override string ToString() => nameof(Inactive);
+
+        public string Identifier => "Inactive";
+    }
+
+    public class Paid : IUserState
+    {
+        public static Paid Instance { get; } = new Paid();
+
+        private Paid() { }
+        public override bool Equals(object obj) => !ReferenceEquals(obj, null) && obj is Paid;
+        public override int GetHashCode() => nameof(Paid).GetHashCode();
+        public override string ToString() => nameof(Paid);
+
+        public string Identifier => "Paid";
```

Comment:
This will be a irritating for people, who are using state types defined in libraries with no Akka dependency. Maybe we could provide something like:

```csharp
interface IIdentity<TState>
{
    string GetIdentifier(TState state);
}
```
It's like `IEqualityComparer<>` (so we can supply our own impl of it even thou each object already implements some GetHashCode and Equals method).